Fix coordinator timer subscription test for learning save interval

## Summary
1 test is failing because coordinator now has two timer subscriptions instead of one.

## Affected Test
- `test_async_start_subscribes_to_events` in `tests/test_coordinator.py`

## Error Message
```
AssertionError: Expected 'async_track_time_interval' to have been called once. Called 2 times.
Calls: [call(..., _handle_periodic_tick, timedelta(seconds=60)),
       call(..., _handle_learning_save, timedelta(seconds=300))]
```

## Root Cause
The coordinator now subscribes to two time intervals:
1. `_handle_periodic_tick` - every 60 seconds
2. `_handle_learning_save` - every 300 seconds (for learning system persistence)

## Fix
Update the test to expect 2 calls to `async_track_time_interval` and verify both handlers are registered.

## Files to Update
- `tests/test_coordinator.py`

## Priority
Medium - test needs to reflect current implementation

--- a/tests/test_coordinator.py
+++ b/tests/test_coordinator.py
@@ -171,8 +171,8 @@ class TestAsyncStart:
 
             # Verify state change subscription
             mock_track_state.assert_called_once()
-            # Verify periodic timer subscription
-            mock_track_time.assert_called_once()
+            # Verify periodic timer subscriptions (periodic tick + learning save)
+            assert mock_track_time.call_count == 2
             # Verify midnight, daily summary, and thermal mode decision subscriptions
             assert mock_track_time_change.call_count == 3
 

